Refine setup and configuration docs (#2082)
open-telemetry/opentelemetry-dotnet-instrumentation: docs/windows-service-instrumentation.md

# Instrument a Windows Service running a .NET application

## Setup

Use the `OpenTelemetry.DotNet.Auto.psm1"` PowerShell module

> to setup automatic instrumentation for a Windows Service:

Reviewer: ```suggestion
to set up automatic instrumentation for a Windows Service:
```

```powershell
# Import the module
Import-Module "OpenTelemetry.DotNet.Auto.psm1"

# Install core files
Install-OpenTelemetryCore

> # Setup your Windows Service instrumentation

Reviewer: ```suggestion
# Set up your Windows Service instrumentation
```

Register-OpenTelemetryForWindowsService -WindowsServiceName "WindowsServiceName" -OTelServiceName "MyServiceDisplayName"
```

⚠️ `Register-OpenTelemetryForWindowsService` performs a service restart.

## Configuration

> Remember to restart the Windows Service after making configuration changes.
> You can do it by executing
> `Restart-Service -Name $WindowsServiceName -Force` in PowerShell.

For .NET Framework application you can configure the most common `OTEL_` settings
(like `OTEL_RESOURCE_ATTRIBUTES`) via `appSettings` in `Web.config`.

The alternative is to set environment variables for the Windows Service
in the Windows Registry.

The registry key of a given Windows Service (named `$svcName`) is located under:

```powershell
HKLM\SYSTEM\CurrentControlSet\Services\$svcName
```

> The environment variables can are defined

Reviewer: ```suggestion
The environment variables are defined
```
